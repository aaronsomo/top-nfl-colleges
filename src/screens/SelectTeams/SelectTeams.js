import React, { useEffect, useState } from 'react';
import { SafeAreaView, StyleSheet, Text, Button, View } from 'react-native';
import axios from 'axios';
import { FlatList, ScrollView, TouchableOpacity } from 'react-native-gesture-handler';

const SelectTeams = ({ navigation }) => {
    const [teams, setTeams] = useState([]);

    const fetchTeams = () => {
        axios.get('http://site.api.espn.com/apis/site/v2/sports/football/nfl/teams')
            // .then(response => console.log(response.data.sports[0].leagues[0].teams));
            .then(({ data }) => (
                // data.sports[0].leagues[0].teams.map(team => {
                //     <Text>{team.displayName}</Text>
                // })
                setTeams(data.sports[0].leagues[0].teams)
            ));
    };

    useEffect(() => {
        fetchTeams();
    }, []);

    return (
        <SafeAreaView style={styles.container}>
            <Text>SelectTeams Page</Text>
            <Text>Testing</Text>
            <View style={{ height: 480 }}>
                <ScrollView style={{ display: 'flex', width: '100%' }}>
                    <View style={{ display: 'flex', flexDirection: 'row', justifyContent: 'center', flexWrap: 'wrap' }}>
                        {teams.length > 0 && teams.map((team, index) => (
                            <TouchableOpacity style={{ color: 'black', height: 100, width: 100, margin: 10, borderWidth: 1, borderColor: 'black', borderRadius: 10 }}>
                                <Button key={index} title={team.team.displayName} />
                            </TouchableOpacity>
                        ))}
                    </View>
                </ScrollView>
            </View>
            <Button
                title="Results"
                onPress={() => navigation.navigate("Results")}
            />
        </SafeAreaView>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
})

export default SelectTeams;