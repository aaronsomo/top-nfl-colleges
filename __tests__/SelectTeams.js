jest.useFakeTimers()
import 'react-native';
import React from 'react';
import SelectTeams from '../src/screens/SelectTeams/SelectTeams';
import { NavigationContainer } from '@react-navigation/native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { fireEvent, render } from 'react-native-testing-library';


describe('<SelectTeams />', () => {
    it('renders correctly', () => {
        renderer.create(
            <NavigationContainer>
                <SelectTeams />
            </NavigationContainer>
        );
    });

    it('should match snapshot',() => {
        const mockFunction = jest.fn();
        const TestSelectTeams = render(<SelectTeams fetchTeamRoster={mockFunction} countColleges={mockFunction} />).toJSON();
        expect(TestSelectTeams).toMatchSnapshot();
    });

    it('navigates to Results screen', () => {
        const navigate = jest.fn();

        const { getByText } = render(
            <NavigationContainer>
                <SelectTeams
                    navigation={{ navigate }}
                />
            </NavigationContainer>
        );
        fireEvent.press(getByText('Scout >>'));
        expect(navigate).toHaveBeenCalled();
    });

});