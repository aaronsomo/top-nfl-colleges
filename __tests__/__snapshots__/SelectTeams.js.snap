// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectTeams /> should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#595faa",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 30,
        "fontWeight": "600",
        "height": 60,
        "textAlign": "center",
        "width": 200,
      }
    }
  >
    Select Teams
  </Text>
  <View
    style={
      Object {
        "height": 480,
      }
    }
  >
    <RCTScrollView
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      style={
        Object {
          "display": "flex",
          "width": "100%",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "70%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3898EC",
            "borderRadius": 10,
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "500",
            }
          }
        >
          Unselect All
        </Text>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3898EC",
            "borderRadius": 10,
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "500",
            }
          }
        >
          Select All
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3898EC",
        "borderRadius": 10,
        "height": 60,
        "justifyContent": "center",
        "margin": 15,
        "width": 120,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d3d3d3",
            "borderRadius": 10,
            "height": 60,
            "justifyContent": "center",
            "margin": 15,
            "opacity": 1,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "500",
            }
          }
        >
          Scout &gt;&gt;
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</RCTSafeAreaView>
`;
